### PR TITLE
feat: tf-idf weighting for pairwise patch similarity in GramAnchoringObjective

### PR DESCRIPTION
**Pull request by sandhawalia**
TF-IDF weight of patches in forward dynamics.

### CODE REVIEW

File: src/rmind/components/loss.py [hunk 1]
```diff
@@ -78,59 +78,62 @@ def forward(
...
+        input_n = F.normalize(input_view, dim=-1)
+        target_n = F.normalize(target_view, dim=-1)
+
+        # Target-driven within-frame patch uniqueness weights.
+        # Patches similar to many others in the same frame are downweighted.
+        frame_sim = torch.einsum("bpd,bqd->bpq", target_n, target_n).clamp_min(0.0)
+        if self.patches == 1:
+            weights = torch.ones_like(frame_sim[..., 0])
+        else:
+            eye = torch.eye(self.patches, dtype=torch.bool, device=target.device)
```
Comment (sandhawalia) [inline src/rmind/components/loss.py]: patches = 1 would this happen ?

Comment (koritsky) [reply src/rmind/components/loss.py]: words of wisdom

**Review by sandhawalia — COMMENTED** [review]
sanity training run and merge ?
